lib/matplotlib/cbook/__init__.py

```diff
@@ -2156,3 +2156,65 @@ def _check_in_list(values, **kwargs):
             raise ValueError(
                 "{!r} is not a valid value for {}; supported values are {}"
                 .format(v, k, ', '.join(map(repr, values))))
+
+
+_valid_compass = ['NE', 'NW', 'SW', 'SE', 'E', 'W', 'E', 'S', 'N', 'C']
+
+
+def _map_loc_to_compass(loc, allowtuple=False, allowbest=False,
+                        fallback='NE', warnonly=True):
+    """
+    Map a location string to a compass notation string. This is used by
+    AnchoredOffsetbox and Legend.
+    """
+    codes = {
+             'upper right':  'NE', 'northeast': 'NE', 1: 'NE',
+             'upper left':   'NW', 'northwest': 'NW', 2: 'NW',
+             'lower left':   'SW', 'southwest': 'SW', 3: 'SW',
+             'lower right':  'SE', 'southeast': 'SE', 4: 'SE',
+             'right':        'E',                     5: 'E',
+             'center left':  'W',  'west':      'W',  6: 'W',
+             'center right': 'E',  'east':      'E',  7: 'E',
+             'lower center': 'S',  'south':     'S',  8: 'S',
+             'upper center': 'N',  'north':     'N',  9: 'N',
+             'center':       'C',                    10: 'C'
+             }
+
+    if allowbest:
+        codes.update({'best': 'best', 0: 'best'})
+
+    if loc in _valid_compass:
+        return loc
+
+    if isinstance(loc, str) or isinstance(loc, int):
+        if loc in codes:
+            return codes[loc]
+
+    if allowtuple:
```

Comment:
In case of a legend `loc` can be a 2-tuple (or any list/array of two numbers). The docstrings states 

> The location can also be a 2-tuple giving the coordinates of the lower-left
    corner of the legend in axes coordinates (in which case *bbox_to_anchor*
    will be ignored).

For any other anchored box, this tuple location is not allowed, so the `allowtuple` flag makes sure that the tuple only makes it into the output in case of a legend.